runtime/astar/src/lib.rs

```diff
@@ -847,6 +847,140 @@ impl pallet_xc_asset_config::Config for Runtime {
     type WeightInfo = pallet_xc_asset_config::weights::SubstrateWeight<Self>;
 }
 
+/// The type used to represent the kinds of proxying allowed.
+#[derive(
+    Copy,
+    Clone,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Encode,
+    Decode,
+    RuntimeDebug,
+    MaxEncodedLen,
+    scale_info::TypeInfo,
+)]
+pub enum ProxyType {
+    /// Allows all runtime calls for proxy account
+    Any,
+    /// Allows only NonTransfer runtime calls for proxy account
+    /// To know exact calls check InstanceFilter inmplementation for ProxyTypes
+    NonTransfer,
+    /// All Runtime calls from Pallet Balances allowed for proxy account
+    Balances,
+    /// All Runtime calls from Pallet Assets allowed for proxy account
+    Assets,
+    /// Only provide_judgement call from pallet identity allowed for proxy account
+    IdentityJudgement,
+    /// Only reject_announcement call from pallet proxy allowed for proxy account
+    CancelProxy,
+    /// All runtime calls from pallet DappStaking allowed for proxy account
+    DappsStaking,
+}
```

Comment:
Can you please extend it with the additional type for reward claiming?

Reply:
will do.